### DISCUSSION TITLE
'Realtime' monitoring of custom summary target?

### DISCUSSION THREAD

**Discussion by stuvet** [ropensci/targets#912] [Ideas]
With the current framework I can write a target to collate summary `tibbles` into a single summary `tibble` with various custom summary statistics (e.g. rhat, rmse, mae etc).

For statically-named targets I currently do this like:

```
tar_monitor<-tarchetypes::tar_eval(
  tar_target(
    monitor_mcmc,
    {
      dplyr::bind_rows(!!!upstream_targets_sym)
    },
    tidy_eval = TRUE
  ),
  values = list(
    upstream_targets_sym = list(rlang_syms(sprintf('fit_monitor_%s', 1:10)))
  )
)

```

It'd be great to be able to make this summary target while the pipeline is still running to allow near real-time monitoring.

Can you suggest any way this might be achieved, collating only the outputs of targets that haven't been marked as 'started' (I can deal with creating the target despite errors using `error = 'null'`, though maybe there'd be a better way).

Thanks

Edit: This seems to have the desired behaviour, though I have to wait for a few thousand 'skipped' targets to cycle through, though I think I could get around this by caching the list of upstream targets to split the two parts of the pipeline.

```
tar_monitor<-tarchetypes::tar_eval(
  tar_target(
    monitor_mcmc,
     dplyr::bind_rows(!!!rlang::syms(upstream_targets_str[upstream_targets_str %in% tar_built()])),
    tidy_eval = TRUE
  ),
  values = list(
    upstream_targets_str = list(sprintf('fit_monitor_%s', 1:10))
  )
)

```

I'm certain it's not clever to run the pipeline in two simultaneous processes, but can you see any downsides of doing it this way? (where I should only ever be skipping targets apart from the last one, not rebuilding them). I guess that target would never fully build at the end of the pipeline, so it'd be better as a `monitor_mcmc_snap`, with the full `monitor_mcmc` as I previously had it.

**Comment by wlandau** [comment]
In your first setup, you could get a mid-pipeline assessment with something like this:

```r
envir <- new.env(parent = globalenv())
tar_load(starts_with("fit_monitor_"), envir = envir)
dplyr::bind_rows(as.list(envir))
```

Depending on the kind of output you want, you could filter out skipped targets using the output from `tar_progress()`.
